src/medCore/gui/toolboxes/medSelectorToolBox.cpp

```diff
@@ -0,0 +1,173 @@
+/*=========================================================================
+
+ medInria
+
+ Copyright (c) INRIA 2013 - 2014. All rights reserved.
+ See LICENSE.txt for details.
+ 
+  This software is distributed WITHOUT ANY WARRANTY; without even
+  the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+  PURPOSE.
+
+=========================================================================*/
+
+#include <medAbstractToolBox.h>
+#include <medSelectorToolBox.h>
+#include <medToolBoxFactory.h>
+#include <medToolBoxHeader.h>
+#include <medToolBoxTab.h>
+
+class medSelectorToolBoxPrivate
+{
+public:
+    medComboBox *chooseComboBox;
+    medAbstractToolBox * currentToolBox;
+    QHash<QString, medAbstractToolBox*> toolBoxes;
+    QVBoxLayout *mainLayout;
+
+    medAbstractData *inputData;
+};
+
+medSelectorToolBox::medSelectorToolBox(QWidget *parent, QString tlbxId) :
+    medToolBox(parent),
+    d(new medSelectorToolBoxPrivate)
+{
+    d->currentToolBox = NULL;
+
+    d->chooseComboBox = new medComboBox;
+    d->chooseComboBox->addItem("Choose algorithm");
+    d->chooseComboBox->setToolTip(tr("Browse through the list of available algorithm"));
+
+    medToolBoxFactory* tbFactory = medToolBoxFactory::instance();
+    int i = 1; //account for the choose Filter item
+    foreach(QString toolboxName, tbFactory->toolBoxesFromCategory(tlbxId))
+    {
+        medToolBoxDetails* details = tbFactory->toolBoxDetailsFromId(toolboxName);
+
+        d->chooseComboBox->addItem(details->name, toolboxName);
+        d->chooseComboBox->setItemData(i,
+                                                   details->description,
+                                                   Qt::ToolTipRole);
```

Comment:
This indentation is very strange, and the elements shouldn't even be on separate lines considering they easily fit on one line.
In general it is not considered proper code to artificially align elements of multiple lines, because as class/function/variable names get changed, the indentation isn't valid anymore and leads to this sort of stuff.